src/components/TopNav/config.ts

```diff
@@ -0,0 +1,50 @@
+export interface NavLink {
+  href: string;
+  text: string;
+  children?: NavLink[];
+}
+
+export const primaryNavLinks: NavLink[] = [
+  { href: "/users", text: "Users" },
+  {
+    href: "/services",
+    text: "Services",
+    children: [
+      {
+        href: "/kliento",
+        text: "Kliento",
+        children: [
+          { href: "/kliento/clients", text: "Clients" },
+          { href: "/kliento/servers", text: "Servers" },
+          { href: "/kliento/overview", text: "Overview" },
+        ],
+      },
+    ],
+  },
+  { href: "/overview", text: "Tech overview" },
+  { href: "/about", text: "About" },
+];
+
+export function findServiceByPath(servicePath: string): NavLink {
+  const normalizedPath = servicePath.startsWith("/") ? servicePath : `/${servicePath}`;
+
+  const servicesItem = primaryNavLinks.find((item) => item.href === "/services");
+
+  if (!servicesItem) {
+    throw new Error("Services navigation item not found in primary navigation");
+  }
+
+  if (!servicesItem.children || servicesItem.children.length === 0) {
+    throw new Error("No services defined in navigation config");
+  }
+
+  const service = servicesItem.children.find(
+    (child) => child.href.toLowerCase() === normalizedPath.toLowerCase(),
+  );
+
+  if (!service) {
+    throw new Error(`Service not found for path: ${servicePath}`);
+  }
+
+  return service;
+}
```

Comment:
I think this can be simplified by iterating a single time, and avoiding the path normalisation (I think it's unnecessary; `KlientoPage.astro` and `kliento/index.astro` must just pass a valid path).

```suggestion
export function findServiceByPath(servicePath: string): NavLink {
  const service = primaryNavLinks
    .flatMap((item) => (item.children ? [item, ...item.children] : [item]))
    .find((item) => item.href.toLowerCase() === servicePath.toLowerCase());

  if (!service) {
    throw new Error(`Service not found for path: ${servicePath}`);
  }

  return service;
}
```

(I haven't tested this code)